provider/source/src/lib.rs

```diff
@@ -486,11 +486,11 @@ pub enum CollationHanDatabase {
     Unihan,
 }
 
-impl std::fmt::Display for CollationHanDatabase {
+impl std::fmt::Display for CollationRootHan {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
         match self {
-            CollationHanDatabase::Implicit => write!(f, "implicithan"),
-            CollationHanDatabase::Unihan => write!(f, "unihan"),
+            CollationRootHan::Implicit => write!(f, "implicit"),
```

Comment:
I think this is what breaks everything this impl is used in filenames

Reply:
I'm going to revert the change: it's used in local filenames *and* icuexport filenames, so it's going to be a mess to fix. This is largely internal, I don't care too much about it.